uberfire-extensions/uberfire-widgets/uberfire-widgets-core/uberfire-widgets-core-client/src/main/java/org/uberfire/ext/widgets/core/client/tree/TreeItem.java

```diff
@@ -26,37 +27,69 @@
 import com.google.gwt.user.client.ui.ComplexPanel;
 import com.google.gwt.user.client.ui.Composite;
 import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.IsWidget;
 import com.google.gwt.user.client.ui.SimplePanel;
-import org.gwtbootstrap3.client.ui.Icon;
-import org.gwtbootstrap3.client.ui.constants.IconType;
 import org.uberfire.ext.widgets.core.client.resources.TreeNavigatorResources;
 
 import static org.uberfire.commons.validation.PortablePreconditions.checkNotNull;
 
 public class TreeItem extends Composite {
```

Comment:
I would add a generic type here in `TreeItem` class, for the children stuff... for example:
`public class TreeItem<I extends TreeItem> extends Composite`
This way methods like `getChildren()` or `getChild(index)` can return the concrete class types, no need to perform casts.